pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -52,11 +52,31 @@ const (
 	deleteClusterconfigCommitMessage  = "Delete commit of cluster configuration; generated by EKS-A CLI"
 )
 
+type GitClient interface {
+	Add(filename string) error
+	Remove(filename string) error
+	Clone(ctx context.Context) error
+	Commit(message string) error
+	Push(ctx context.Context) error
+	Pull(ctx context.Context, branch string) error
+	Init() error
+	Branch(name string) error
+}
+
+type GitProviderClient interface {
+	GetRepo(ctx context.Context) (repo *git.Repository, err error)
+	CreateRepo(ctx context.Context, opts git.CreateRepoOpts) (repo *git.Repository, err error)
+	DeleteRepo(ctx context.Context, opts git.DeleteRepoOpts) error
+	Validate(ctx context.Context) error
+	PathExists(ctx context.Context, owner, repo, branch, path string) (bool, error)
+}
+
 type FluxAddonClient struct {
-	flux    Flux
-	writer  filewriter.FileWriter
-	git     git.Provider
-	retrier *retrier.Retrier
+	flux              Flux
+	writer            filewriter.FileWriter
+	gitClient         GitClient
```

Comment:
Wonder what's the reason to split the git provider from the FluxClient level?

Reply:
What code needs a `GitClient`? Does the `GitClient` represent a single repository (seems that way)? If so, what does a `FluxAddonClient` represent? Seems odd that a `FluxAddonClient` would have a reference to a specific repository but also be able to create other repositories in some capacity (SRP)?